lib/active_storage/analyzer/custom_image_analyzer.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module ActiveStorage
+  class Analyzer::CustomImageAnalyzer < Analyzer::ImageAnalyzer
+    def self.accept?(blob)
+      ['image/jpeg', 'image/pjpeg'].include?(blob.content_type) && blob.attachments.take.record_type == 'Asset'
```

Comment:
Would it make sense to make this array an constant? And maybe use activesupport .in? (seems more readable)